controllers/keystoneapi_controller.go

```diff
@@ -225,10 +228,17 @@ func (r *KeystoneAPIReconciler) reconcileInit(
 		},
 	)
 	// create or patch the DB
+	// TODO1 pass the reconciler and api instance objects to createDb.
 	ctrlResult, err := db.CreateOrPatchDB(
 		ctx,
+		instance,
 		helper,
 	)
+	// Register the finalizer immediately to avoid orphaning resources on delete
+	if err := r.Update(ctx, instance); err != nil {
+		return ctrl.Result{}, err
+	}
+
```

Comment:
see  my comment on the other PR. I don't think this is needed

Reply:
ok